src/main/java/mekanism/common/tile/TileEntityDimensionalStabilizer.java

```diff
@@ -0,0 +1,211 @@
+
+    public static final int MAX_LOAD_RADIUS = 2;
+    public static final int MAX_LOAD_DIAMETER = 2 * MAX_LOAD_RADIUS + 1;
+
+    protected final ChunkLoader chunkLoaderComponent;
+    protected final boolean[][] loadingChunks;
+
+    protected boolean chunkloading = false;
+
+    protected FixedUsageEnergyContainer<TileEntityDimensionalStabilizer> energyContainer;
+    @WrappingComputerMethod(wrapper = ComputerIInventorySlotWrapper.class, methodNames = "getEnergyItem")
+    private EnergyInventorySlot energySlot;
+
+    public TileEntityDimensionalStabilizer(BlockPos pos, BlockState state) {
+        super(MekanismBlocks.DIMENSIONAL_STABILIZER, pos, state);
+
+        chunkLoaderComponent = new ChunkLoader(this);
+        loadingChunks = new boolean[MAX_LOAD_DIAMETER][MAX_LOAD_DIAMETER];
+        loadingChunks[MAX_LOAD_RADIUS][MAX_LOAD_RADIUS] = true;
+    }
+
+    public boolean isChunkloadingAt(int x, int z) {
+        return loadingChunks[x][z];
+    }
+
+    public void toggleChunkloadingAt(int x, int z) {
+        loadingChunks[x][z] = !loadingChunks[x][z];
+        setChanged(false);
+    }
+
+    @Nonnull
+    @Override
+    protected IEnergyContainerHolder getInitialEnergyContainers(IContentsListener listener) {
+        EnergyContainerHelper builder = EnergyContainerHelper.forSide(this::getDirection);
+        builder.addContainer(energyContainer = FixedUsageEnergyContainer.input(this, listener));
+        return builder.build();
+    }
+
+    @Nonnull
+    @Override
+    protected IInventorySlotHolder getInitialInventory(IContentsListener listener) {
+        InventorySlotHelper builder = InventorySlotHelper.forSide(this::getDirection);
+        builder.addSlot(energySlot = EnergyInventorySlot.fillOrConvert(energyContainer, this::getLevel, listener, 143, 35), RelativeSide.BACK);
+        return builder.build();
+    }
+
+    @Override
+    public void addContainerTrackers(MekanismContainer container) {
+        super.addContainerTrackers(container);
+        container.trackArray(loadingChunks);
+    }
+
+    @Override
+    protected void onUpdateServer() {
+        super.onUpdateServer();
+        energySlot.fillContainerOrConvert();
+        if (MekanismUtils.canFunction(this)) {
+            FloatingLong energyPerTick = energyContainer.getEnergyPerTick().multiply(getChunksLoaded());
+            if (energyContainer.extract(energyPerTick, Action.SIMULATE, AutomationType.INTERNAL).equals(energyPerTick)) {
+                energyContainer.extract(energyPerTick, Action.EXECUTE, AutomationType.INTERNAL);
+                setChunkloading(true);
+            } else {
+                setChunkloading(false);
+            }
+        }
+    }
+
+    private void setChunkloading(boolean value) {
+        boolean old = chunkloading;
+        chunkloading = value;
+        if (old != chunkloading) {
+            setChanged(true);
+        }
+    }
+
+    private int getChunksLoaded() {
```

Comment:
This method might also be worth exposing to computers, though I wonder if the one exposed to them should be wrapped basically as:
```java
return chunkLoading ? getChunksLoaded() : 0;
```